collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -361,21 +354,16 @@ func (r *Receiver) anyStream(serverStream anyStreamServer, method string) (retEr
 		// Receive a batch corresponding with one ptrace.Traces, pmetric.Metrics,
 		// or plog.Logs item.
 		req, err := serverStream.Recv()
-
 		if err != nil {
-			// client called CloseSend()
+			// This includes the case where a client called CloseSend(), in
+			// which case we see an EOF error here.
+			r.logStreamError(err)
+
 			if errors.Is(err, io.EOF) {
-				status := &arrowpb.BatchStatus{}
-				status.StatusCode = arrowpb.StatusCode_CANCELED
-				err = serverStream.Send(status)
-				if err != nil {
-					r.logStreamError(err)
-					return err
-				}
-				return nil
+				return status.Error(codes.Canceled, "client stream shutdown")
+			} else if errors.Is(err, context.Canceled) {
+				return status.Error(codes.Canceled, "server stream shutdown")
```

Comment:
This is the big change. Instead of returning an error, returning a status.Error() w/ the canceled code -- now the exporter will see that canceled in its reader and shut down cleanly.